proposals/0759-react-native-frameworks.md

```diff
@@ -0,0 +1,452 @@
+- Will be mentioned during public comms as one of the recommended frameworks by the React Native core team.
+- The company/entity behind the Framework will automatically be granted a [partner status](https://github.com/facebook/react-native/blob/main/ECOSYSTEM.md#partners).
+- PRs and Issues coming from the Framework developers will receive higher priorities.
+
+### Requirements
+
+Here is a short summary of what are the requirements for React Native Frameworks.
+
+#### DOs of Frameworks
+
+- **Focus on developer & user experience** - We believe Frameworks should aim to provide a top-notch developer experience. Frameworks that fail to provide good documentation or developer workflows are not likely high quality frameworks, and our users will most likely not want to use them. Similarly, we want to promote frameworks that provide a top-notch user experience and do not ignore basic user needs, such as accessibility, performance, and delightful UIs.
+- **Focus on building cohesive experiences** - Users are looking into deploying universal apps across multiple targets. We’re delegating this responsibility to frameworks that can provide better experiences to users and craft cohesive developer experiences between mobile, web, desktop and beyond.
+- **Focus on abstracting complexity away** - Any initiative that simplifies the developer workflow is welcomed in the framework domain.
+- **Be popular** - We’ll only be recommending solutions that are widely known and adopted by the community.
+- **Open source** - Recommended frameworks must be free, open source, projects. We allow for companies to provide paid services on top of the Open Source framework, but these services should not be a requirement to use the framework and should not have vendor lock-in.
+
+#### DON’Ts of Frameworks
+
+- **Do not hard fork react-native** - As a rule of thumb, you should not GitHub-fork react-native and re-publish it. If you’re doing so, it’s probably due to a lack of APIs that React Native should provide. Please reach out to us and attempt to de-fork your framework.
+- **Do not soft fork the core APIs** - We want to make sure developers need to learn only once how to write a native component. Building abstraction and wrappers is accepted, forking is not, as it makes it harder for users to share components/modules between targets.
+- **Do not fork the ecosystem** - It’s crucial that `react-native-*` libraries can be easily shared and ported to other targets or frameworks (with little to no work). Frameworks should aim at building libraries that work across the whole ecosystem and not solutions that silo the users away.
+- **Do not redistribute prebuilts** - React Native ships native prebuilts for several targets. Do not attempt to republish them as this reduces security for users and is essentially yet another fork.
```

Comment:
Relatedly, I believe the static library for Hermes will need to be signed by Meta for the iOS in the future to secure part of the SBOM (app review policy). Probably too tangential of a detail to mention in the RFC, but it is another reason that reinforces this point here.

Reply:
Hi @ide, sorry for the late reply. For the privacy updates Apple is pushing, we have an internal effort to fix them. 
For most of them, we will include a privacy manifest plist in the template, which Frameworks could take as starting point if they use other restricted APIs.

For `hermes` specifically, we are talking with Apple directly to understand what to do. We might end up removing/replacing the offending methods to avoid to sign the binary.